lib/upgrade/upgrade_supported.go

```diff
@@ -205,7 +205,12 @@ func upgradeToURL(archiveName, binary string, url string) error {
 	if err != nil {
 		return err
 	}
-	return os.Rename(fname, binary)
+	if os.Rename(fname, binary); err != nil {
```

Comment:
Needs removing the temp file in the error return above.  Maybe defer a Remove(fname) and just do the rename you do here?